include/deal.II/base/parameter_handler.h

```diff
@@ -125,6 +125,90 @@ namespace Patterns
      * assumption significantly, you can still overload this function.
      */
     virtual std::size_t memory_consumption () const;
+
+
+    /**
+     * Convert a compatible value type to a string. If the type is not
+     * compatible, an excepion ExcIncompatibleType is thrown.
+     *
+     * Internally, this function wraps the value in a boost::any object
+     * and calls the virtual function any_to_string(). Derived patterns
+     * should implement any_to_string for this function to work properly.
+     *
+     * An example usage is here:
+     *
+     * @code
+     * int a = 5;
+     * Patterns::Integer p;
+     *
+     * std::string s = p.to_string(a); // s = "5"
+     * @endcode
+     */
+    template<class T>
+    std::string to_string(const T &value) const;
+
+
+    /**
+     * Translate the content of the string matching this pattern, into a
+     * value and store it in @p value.  If the type T is not
+     * compatible with the pattern, an excepion ExcIncompatibleType is
+     * thrown.
+     *
+     * Internally, this function wraps a pointer to @p value in a
+     * boost::any object and calls the virtual function string_to_any(). Derived
+     * patterns should implement string_to_any for this function to work properly.
+     *
+     * An example usage is here:
+     *
+     * @code
+     * int a;
+     * Patterns::Integer p;
+     *
+     * p.to_value("5", a);
+     * @endcode
+     */
+    template<class T>
+    void to_value(const std::string &s, T &value) const;
```

Comment:
@tamiko , you mean
~~~
const auto value = [...].to_value<T>(string);
~~~
instead of 
~~~
[...].to_value(string, value);
~~~
where `<T>` is not needed. Inferring the type here is useless, since this function is templated on the type `T`, so you have to specify it anyway (unless you pass it as a parameter).